src/neo4j_genai/types.py

```diff
@@ -93,3 +99,59 @@ class SearchType(str, Enum):
 
     VECTOR = "vector"
     HYBRID = "hybrid"
+
+
+class EmbedderModel(BaseModel):
+    embedder: Optional[Any]
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
I probably missed something but where are these `model_config` used?

Reply:
Good question because it's subtle. Without `model_config` we will get errors when trying to pass arbitrary types like `Embedder`. I chose to pass this into the `EmbedderModel` instead of every retriever data model for better encapsulation

[In Pydantic V2 you can do this as seen here](https://docs.pydantic.dev/2.7/api/config/#pydantic.config.ConfigDict.arbitrary_types_allowed)

Reply:
Got it, thanks